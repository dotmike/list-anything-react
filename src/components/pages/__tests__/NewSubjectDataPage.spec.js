import React from "react";
import configureStore from "redux-mock-store";

import ConnectedNewSubjectDataPage, {
  UnconnectedNewSubjectDataPage
} from "../NewSubjectDataPage";

const defaultProps = {
  history: {
    push: () => {}
  },
  match: {
    params: {
      _id: "1"
    }
  }
};

describe("ConnectedNewSubjectDataPage", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <ConnectedNewSubjectDataPage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedNewSubjectDataPage", () => {
  const props = {
    ...defaultProps,
    subjectId: "",
    createSubjectData: () => Promise.resolve(),
    fetchSubjectById: () => Promise.resolve()
  };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <UnconnectedNewSubjectDataPage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
