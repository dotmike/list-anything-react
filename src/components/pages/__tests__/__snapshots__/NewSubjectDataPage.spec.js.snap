// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedNewSubjectDataPage renders correctly 1`] = `
<NewSubjectDataPage
  createSubjectData={[Function]}
  fetchSubjectById={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "_id": "1",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  subjectId=""
/>
`;

exports[`UnconnectedNewSubjectDataPage renders correctly 1`] = `
<Segment
  style={
    Object {
      "margin": "10px auto",
      "maxWidth": 500,
    }
  }
>
  <Header
    as="h2"
    color="teal"
    textAlign="center"
  >
    Add new Subject Data
  </Header>
  <Loader
    active={true}
    inline="centered"
  />
</Segment>
`;
